fix startup error

**Describe the bug**
When starting up release `2.3.1` in docker the following error is thrown: 

```
senzing-webapp   |
senzing-webapp   | > @senzing/entity-search-web-app@2.3.1 start:docker /app
senzing-webapp   | > node ./run/webserver
senzing-webapp   |
senzing-webapp   | internal/modules/cjs/loader.js:892
senzing-webapp   |   throw err;
senzing-webapp   |   ^
senzing-webapp   |
senzing-webapp   | Error: Cannot find module './utils'
senzing-webapp   | Require stack:
senzing-webapp   | - /app/run/runtime.datastore.js
senzing-webapp   | - /app/run/webserver/index.js
senzing-webapp   |     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:889:15)
senzing-webapp   |     at Function.Module._load (internal/modules/cjs/loader.js:745:27)
senzing-webapp   |     at Module.require (internal/modules/cjs/loader.js:961:19)
senzing-webapp   |     at require (internal/modules/cjs/helpers.js:92:18)
senzing-webapp   |     at Object.<anonymous> (/app/run/runtime.datastore.js:4:65)
senzing-webapp   |     at Module._compile (internal/modules/cjs/loader.js:1072:14)
senzing-webapp   |     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
senzing-webapp   |     at Module.load (internal/modules/cjs/loader.js:937:32)
senzing-webapp   |     at Function.Module._load (internal/modules/cjs/loader.js:778:12)
senzing-webapp   |     at Module.require (internal/modules/cjs/loader.js:961:19) {
senzing-webapp   |   code: 'MODULE_NOT_FOUND',
senzing-webapp   |   requireStack: [ '/app/run/runtime.datastore.js', '/app/run/webserver/index.js' ]
senzing-webapp   | }
senzing-webapp   | npm ERR! code ELIFECYCLE
senzing-webapp   | npm ERR! errno 1
senzing-webapp   | npm ERR! @senzing/entity-search-web-app@2.3.1 start:docker: `node ./run/webserver`
senzing-webapp   | npm ERR! Exit status 1

```

**Expected behavior**
it start up fine

**Additional context**
this is because the `/run` path is excluded from `git commit` paths by default, `-f` needs to be used to add files under that directory.

--- a/run/runtime.datastore.config.js
+++ b/run/runtime.datastore.config.js
@@ -394,7 +394,7 @@ function getWebServerOptionsFromInput() {
     retOpts.protocol              = cmdLineOpts.protocol ?              cmdLineOpts.protocol              : retOpts.protocol;
     retOpts.port                  = cmdLineOpts.webServerPortNumber ?   cmdLineOpts.webServerPortNumber   : retOpts.port;
     retOpts.hostname              = cmdLineOpts.webServerHostName ?     cmdLineOpts.webServerHostName     : retOpts.hostname;
-    retOpts.url                   = cmdLineOpts.url ?                   cmdLineOpts.webServerUrl          : retOpts.url;
+    retOpts.url                   = cmdLineOpts.webServerUrl ?          cmdLineOpts.webServerUrl          : retOpts.url;
     retOpts.apiPath               = cmdLineOpts.webServerApiPath ?      cmdLineOpts.webServerApiPath      : retOpts.apiPath;
     retOpts.authPath              = cmdLineOpts.webServerAuthPath ?     cmdLineOpts.webServerAuthPath     : retOpts.authPath;
     retOpts.authMode              = cmdLineOpts.webServerAuthMode ?     cmdLineOpts.webServerAuthMode     : retOpts.authMode;
@@ -421,7 +421,17 @@ function getWebServerOptionsFromInput() {
   }
   // if we just have "hostname" construct "url" from that
   retOpts.url                   = !retOpts.url && retOpts.hostname ? ((retOpts.protocol ? retOpts.protocol + '://' : 'http://') + retOpts.hostname +(retOpts.port ? ':'+ retOpts.port : '')) : retOpts.url;
-
+  let hostnameFromUrL           = getHostnameFromUrl(retOpts.url);
+  let portFromUrl               = getPortFromUrl(retOpts.url);
+  
+  if(hostnameFromUrL && hostnameFromUrL !== retOpts.hostname){
+    // override from url
+    retOpts.hostname            = hostnameFromUrL;
+  }
+  if(portFromUrl && portFromUrl !== retOpts.port){
+    // override from url
+    retOpts.port                = portFromUrl;
+  }
   return retOpts;
 }
 

